File: 2-Patron/php/Armadura.php
<?php
class Armadura
{
  private $nombre;
  private $danoRecibido;
  private $nivel;
  private $duracion;

  function setNombre()
  {
    return $this->nombre;
  }

  function getNombre($nNombre)
  {
    $this->nombre = $nNombre;
  }

  function setDanoRecibido()
  {
    return $this->danoRecibido;
  }

  function getDanoRecibido($nDano)
  {
    $this->danoRecibido = $nDano;
  }

  function setNivel()
  {
    return $this->nivel;
  }

  function getNivel($nNivel)
  {
    $this->nivel = $nNivel;
  }

  function setDuracion()
  {
    return $this->duracion;
  }

  function getDuracion($nDuracion)
  {
    $this->duracion = $nDuracion;
  }

  function incDañoRecibido($incremento)
  {
    $this->danoRecibido += $incremento;
  }
  function incDuracion($incremento)
  {
    $this->duracion += $incremento;
  }
  function incNivel()
  {
    $this->nivel += 1;
  }
}
?>

File: 2-Patron/Java/Armadura.java
package abstractmasterwhole;
public class Armadura {
    private String nombre;
    private int nivel; 
    private float dano_recibido;
    private float duracion; 
    
    public Armadura(String nombre, int nivel, float dano_recibido, float duracion){
        this.dano_recibido = dano_recibido;
        this.duracion = duracion;
        this.nivel = nivel;
        this.nombre = nombre;
    }
    public String nombre(){
        return this.nombre;
    }
    public int getNivel(){
        return this.nivel;
    }
    public float getDano_recibido(){
        return this.dano_recibido;
    }
    public float getDuracion(){
        return this.duracion;
    }
    public void setNombre(String nombre){
        this.nombre = nombre;
    }
    public void setNivel(int nivel){
        this.nivel = nivel;
    }
    public void setDano_recibido(float dano_recibido){
        this.dano_recibido += dano_recibido;
    }
    public void setDuracion(float duracion){
        this.duracion = +duracion;  
    }
    public void incNivel(){
        this.nivel += 1;
    }
    public void incDuracion(int duracion){
        this.duracion = duracion;
    }
}
